data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/EntryConfig.java

```diff
@@ -5,22 +5,36 @@
 
 package org.opensearch.dataprepper.plugins.processor.configuration;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
 import jakarta.validation.constraints.NotEmpty;
+import org.opensearch.dataprepper.plugins.processor.GeoIPDatabase;
+import org.opensearch.dataprepper.plugins.processor.GeoIPField;
 
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class EntryConfig {
-    static final String DEFAULT_TARGET = "geolocation";
+    static final String DEFAULT_TARGET = "geo";
     @JsonProperty("source")
     @NotEmpty
     private String source;
 
     @JsonProperty("target")
     private String target = DEFAULT_TARGET;
 
-    @JsonProperty("fields")
-    private List<String> fields;
+    @JsonProperty("include_fields")
+    private List<String> includeFields = new ArrayList<>();
+
+    @JsonProperty("exclude_fields")
+    private List<String> excludeFields = new ArrayList<>();
```

Comment:
I didn't want to deserialize to enum directly because processor doesn't have control of Object Mapper to ignore invalid enum option. It throws an exception. So I kept it as List<String> without any default value. If it's empty no fields will be excluded and all fields are added.